base/compiler/optimize.jl

```diff
@@ -895,12 +897,11 @@ function check_inconsistentcy!(sv::PostOptAnalysisState, scanner::BBScanner)
             any_non_boundscheck_inconsistent || continue
         elseif isa(stmt, ReturnNode)
             sv.all_retpaths_consistent = false
-        else isa(stmt, GotoIfNot)
+        elseif isa(stmt, GotoIfNot)
```

Comment:
Both this change and the one on L530 are critical. I attempted to make tests for these  cases, but in the end, I couldn't make it. AFAIU, the `while !isempty(stmt_ip)` condition at L884 returns `false` always, suggesting that this whole part may be dead code. So I guess there's a deeper logical error within the current algorithm.